source/material_model/rheology/elasticity.cc

```diff
@@ -404,16 +619,31 @@ namespace aspect
       double
       Elasticity<dim>::
       calculate_viscoelastic_strain_rate(const SymmetricTensor<2,dim> &strain_rate,
-                                         const SymmetricTensor<2,dim> &stress,
+                                         const SymmetricTensor<2,dim> &stress_0_advected,
+                                         const SymmetricTensor<2,dim> &stress_old,
+                                         const double viscosity_pre_yield,
                                          const double shear_modulus) const
       {
         // The second term in the following expression corresponds to the
         // elastic part of the strain rate deviator. Note the parallels with the
         // viscous part of the strain rate deviator,
         // which is equal to 0.5 * stress / viscosity.
-        const SymmetricTensor<2,dim> edot_deviator = deviator(strain_rate) + 0.5*stress /
-                                                     calculate_elastic_viscosity(shear_modulus);
-
+        // in.composition contains $\tau^{0}$, so that we can compute
+        // $\dot\varepsilon_T + \frac{\tau^{0adv}}{2 \eta_{el}}$.
+        // During assembly in timestep 0, get_timestep() returns 0. Therefore we have to make an estimated guess
+        // using the maximum timestep parameters capped by the elastic timestep.
+        double dtc = this->get_timestep();
+        if (this->get_timestep_number() == 0 && this->get_timestep() == 0)
+          dtc = std::min(std::min(this->get_parameters().maximum_time_step, this->get_parameters().maximum_first_time_step), elastic_timestep());
+        const double timestep_ratio = dtc / elastic_timestep();
+        const double elastic_viscosity = timestep_ratio * calculate_elastic_viscosity(shear_modulus);
+        const double creep_viscosity = viscosity_pre_yield; //timestep_ratio *  calculate_viscoelastic_viscosity(viscosity_pre_yield, shear_modulus);
```

Comment:
What about this commented code? If not necessary please remove. If this is some sort of explanation, please move to a separate line and expand the explanation.

Reply:
Yep, should be deleted.